presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeWindowQueries.java

```diff
@@ -36,38 +41,71 @@ protected void createTables()
 
     private static final List<String> OVER_CLAUSES_WITHOUT_ORDER_BY = Arrays.asList(
-            "PARTITION BY orderkey",
             "PARTITION BY custkey, orderkey",
-            "PARTITION BY orderdate",
-            "PARTITION BY orderkey, orderdate",
-            "PARTITION BY custkey, orderkey, shippriority",
-            "PARTITION BY orderkey, custkey");
+            "PARTITION BY orderdate, orderkey");
+
+    private static final List<String> FRAME_CLAUSES = Arrays.asList(
+            // Frame clauses in RANGE mode.
+            "RANGE UNBOUNDED PRECEDING",
+            "RANGE CURRENT ROW",
+            "RANGE BETWEEN UNBOUNDED PRECEDING and UNBOUNDED FOLLOWING",
+
+            // Frame clauses in ROWS mode.
+            "ROWS UNBOUNDED PRECEDING",
+            "ROWS BETWEEN CURRENT ROW AND UNBOUNDED FOLLOWING",
+            "ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING",
+            "ROWS BETWEEN 5 PRECEDING AND CURRENT ROW",
+            "ROWS BETWEEN CURRENT ROW AND 5 FOLLOWING",
+            "ROWS BETWEEN orderkey PRECEDING AND CURRENT ROW",
+            "ROWS BETWEEN CURRENT ROW AND orderkey FOLLOWING",
 
-    protected List<String> getRankingQueries(String rankingFunction, boolean orderBy)
+            // Frame clauses with empty frames.
+            "ROWS BETWEEN UNBOUNDED PRECEDING AND 1 PRECEDING",
+            "ROWS BETWEEN 1 PRECEDING AND 4 PRECEDING",
+            "ROWS BETWEEN 1 FOLLOWING AND UNBOUNDED FOLLOWING",
+            "ROWS BETWEEN 4 FOLLOWING AND 1 FOLLOWING");
+
+    protected List<String> getQueries(String function, FunctionType functionType)
     {
         ImmutableList.Builder<String> queries = ImmutableList.builder();
-        List<String> columnProjections = Arrays.asList("orderkey, orderdate, totalprice");
-        List<String> overClauses = orderBy ? OVER_CLAUSES_WITH_ORDER_BY : OVER_CLAUSES_WITHOUT_ORDER_BY;
+        List<String> overClauses = new ArrayList<>(OVER_CLAUSES_WITH_ORDER_BY);
+        List<String> frameClauses = FRAME_CLAUSES;
+        if (functionType == FunctionType.VALUE) {
+            overClauses.addAll(OVER_CLAUSES_WITHOUT_ORDER_BY);
+        }
+        List<String> windowClauseList = new ArrayList<>();
+        String windowClause = new String();
+        int count = 0;
+        final int QUERY_GROUP_SIZE = 5;
 
-        for (String columnProjection : columnProjections) {
-            for (String overClause : overClauses) {
-                queries.add(String.format("SELECT %s, %s OVER (%s) AS rnk FROM orders", columnProjection, rankingFunction, overClause));
+        for (String overClause : overClauses) {
+            for (String frameClause : frameClauses) {
+                count++;
+                windowClause += String.format("%s OVER (%s %s)", function, overClause, frameClause);
+                if (count == QUERY_GROUP_SIZE) {
+                    windowClauseList.add(windowClause);
```

Comment:
Since we are adding to clauseList only after encountering 5 clauses, it means that the last %5 clauses will not get added to the testing if  frameClauses is not a multiple of 5. Please fix.

Reply:
Oh right, sorry for missing that, fixed.